src/main/scala/io/iohk/ethereum/vm/ProgramState.scala

```diff
@@ -117,7 +117,7 @@ case class ProgramState[W <: WorldStateProxy[W, S], S <: Storage[S]](
   def toResult: ProgramResult[W, S] =
     ProgramResult[W, S](
       returnData,
-      if (error.isDefined) 0 else gas,
+      if (error.exists(_.useWholeGas)) 0 else gas,
```

Comment:
I am also not sure about createOp code, https://github.com/mmrozek/mantis/blob/f8db59a8e67d0d1175559a8c2f603494c6ab1e08/src/main/scala/io/iohk/ethereum/vm/OpCode.scala#L707 should we also retrun old word here in case of revert ? Also should gas be consumed here ?  It am not sure it would be good to check other client, also it would be good to run at least part of ets tests and check what is required semantics here.